import wollok.lang.Exception

class MyException inherits Exception {}
class MySubclassException inherits MyException {}
class A { method m1() { throw new MySubclassException() } }

test "testX" {

	const a = new A()
	var result = 0

	try 
		a.m1()

	catch e : MySubclassException
		result = 3

	catch e : MyException
		result = 2

	assert.that(3 == result)


}